### DISCUSSION TITLE
The difference of MO-coeff matrix in kHF exceed 10^-4 between different calculations

### DISCUSSION THREAD

**Discussion by NhanTran120897** [pyscf/pyscf#2634] [Q&A] [answered]
Hello all developers of Pyscf
I have calculated the MO-coeff matrix by kHF of diamond, some elements appear different at each time of calculation, the difference can be higher than 10^-4. The difference of ERI in kmp2 can be observed but the energy of kmp2 or kHF still get the same value. Can someone please tell me why the difference exist and why the energy of kmp2 and kHF unchanged. Thank you!!! 

This is my input file

import pyscf.pbc.tools.pyscf_ase as pyscf_ase
import pyscf.pbc.gto as pbcgto
from pyscf.pbc import gto, scf, cc, mp
from ase.lattice import bulk

c = bulk('C', 'diamond', a=3.5668)
print(c.get_volume())

cell = pbcgto.Cell()
cell.atom = pyscf_ase.ase_atoms_to_pyscf(c)
cell.a = c.cell

cell.basis = 'gth-dzv'
cell.pseudo = 'gth-pade'
cell.verbose = 5
cell.build(None,None)

kpts = cell.make_kpts([1,2,1], scaled_center=[0., 0., 0.])
kmf = scf.KRHF(cell, kpts)
kmf.kernel()
print(kmf.mo_coeff)

mypt = mp.KMP2(kmf)
mypt.kernel()

**Comment by tberkel** [comment, marked as answer]
There is probably a degenerate set of MOs. If MOs are degenerate, then any linear combination is also a set of canonical MOs. So the mo_coeff matrix will be different, as will the ERIs, but the observables (such as the energy) will be the same.

**Comment by NhanTran120897** [comment]
Thank you for your wise and fast answer, I will try with different materials

**Comment by obackhouse** [comment]
a sanity check would be to run this without threading (typically `OMP_NUM_THREADS=1 python ...`), and it should be deterministic -- but as @tberkel says, the degeneracy means that there is nothing incorrect about the indeterminacy that is introduced with threading.
